### ISSUE TITLE
ApplicationError Exception Formatting

### ISSUE THREAD

**Issue by bitbyt3r** [crossbario/autobahn-python#784]
If I throw an exception in a called function the other host prints the unicode version of ApplicationError, which includes the traceback thrown by the remote function interleaved with newlines. If appears the __unicode__ function on ApplicationError expects the traceback to be a list of lines, but it is receiving a string. (As seen here: https://github.com/crossbario/autobahn-python/blob/master/autobahn/wamp/exception.py#L241) As far as I can tell this originates from https://github.com/twisted/twisted/blob/68d2afa5c2776c242d6bb942813b293b027b837c/src/twisted/python/failure.py#L541

Is there a place I should be overriding this behavior, or is this unintentional? I could really use a guide on error handling with autobahn in general.

**Comment by rharding13** [comment]
I'm having this same issue.

I dug into the code history a bit, and found this:

- The traceback support was originally added in [commit 98e0155cfe27c929d645feb62749c3d71f2b43e0 ("improve logging of app errors; allow forwarding of app tracebacks")](https://github.com/crossbario/autobahn-python/commit/98e0155cfe27c929d645feb62749c3d71f2b43e0) by @oberstet.  He added a `tb` argument to `_message_from_exception()`, which is documented as requiring a `list` value.
- The code became broken in [commit e09df6d80abe2b5097b7825f8c3315d03f3037a9 "more fixes and moving to txaio logging"](https://github.com/crossbario/autobahn-python/commit/e09df6d80abe2b5097b7825f8c3315d03f3037a9) by @meejah.  Among many other changes, it now passes the traceback as a single string rather than a list of strings. The documented requirement for a `list` argument to `_message_from_exception()` did not change, nor did the code inside `class ApplicationError` that depends on that value.

I don't understand the codebase well enough to know whether there was a good reason for this, or if it was just an oversight.

**Comment by oberstet** [comment]
thanks a lot for digging into that! it sounds like we evolved the code on 2 ends (autobahn and txaio) - and broke it on the way;) I think txaio should be in the lead, drive that. if so, then yeah, autobahn would need to adjust. @meejah any opinions? could you have a look?

**Comment by rharding13** [comment]
I think the easiest fix would be to put back in the `splitlines()` that got dropped.  But I don't know if there's code somewhere else (other than `class ApplicationError`) that is now dependant on the single-string-with-embedded-newlines value.

**Comment by meejah** [comment]
Yeah, I think letting `txaio` format the tracebacks is probably best; see PR.
